util/deephash/deephash.go

```diff
@@ -350,6 +358,18 @@ func hashAddr(h *hasher, p pointer) {
 	}
 }
 
+func makeSelfHasherImpl(t reflect.Type) typeHasherFunc {
+	return func(h *hasher, p pointer) {
+		e := p.asValue(t)
+		if t.Kind() == reflect.Pointer && e.IsNil() {
+			h.HashUint8(0) // indicates nil
+			return
+		}
```

Comment:
Is L364 to 367 needed?

The documentation for typeHasherFunc says that:
> The provided pointer must always be non-nil.

Reply:
Also, I don't think `t.Kind()` will ever be `reflect.Pointer` since we have the `t.Kind() != reflect.Pointer` conditional check above.